### DISCUSSION TITLE
runlevel for desktop mode, server mode, etc.

### DISCUSSION THREAD

**Discussion by leleleSDX** [OpenRC/openrc#621] [Q&A] [answered]
how do I use the runlevel system to make init scripts start when there is a mode i want to boot into?

- move all the dekstop essential scripts to the desktop runlevel
- server runlevel can have smbd, docker, squid etc
- can boot directly into the server runlevel to be a headless server
- boot into desktop mode, start the server runlevel without exiting desktop mode so i can do configurations

**Comment by vapier** [comment]
openrc has supported this kind of setup forever.  runlevels even stack.

https://github.com/OpenRC/openrc/blob/master/user-guide.md#runlevels

wrt stacking, create something like `/etc/runlevels/some-common-name/`, put services in there, and then symlink the runlevel into a different one.  so create `/etc/runlevels/foo/` and then create `/etc/runlevels/foo/some-common-name -> ../some-common-name`.

as for switching, use softlevel= kernel command line option in your boot config (e.g. grub) to select the initial one.  and run `openrc foo` to switch at runtime.

**Comment by leleleSDX** [comment]
so it's safe to move desktop required inits to a "desktop" runlevel? how do I make sure that all default inits run when I change the boot into the "desktop" runlevel?

**Reply by vapier** [reply, marked as answer]
that's up to you to make sure you symlink all the required runlevels/services you care about
